code/modules/surgery/organs/lavaland.dm

```diff
@@ -87,12 +109,78 @@
 			L.throw_at_fast(throwtarget, 3, 1)
 		next_command = world.time + cooldown_damage
 
+	//WALK
+	else if(findtext(command, "walk"))
+		for(var/mob/living/L in listeners)
+			L.m_intent = MOVE_INTENT_WALK
+		next_command = world.time + cooldown_meme
+
+	//RUN
+	else if(findtext(command, "run"))
+		for(var/mob/living/L in listeners)
+			L.m_intent = MOVE_INTENT_RUN
+		next_command = world.time + cooldown_meme
+
+	//HELP INTENT
+	else if(findtext(command, "help"))
+		for(var/mob/living/L in listeners)
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				H.a_intent_change(INTENT_HELP)
+		next_command = world.time + cooldown_meme
+
+	//DISARM INTENT
+	else if(findtext(command, "disarm"))
+		for(var/mob/living/L in listeners)
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				H.a_intent_change(INTENT_DISARM)
+		next_command = world.time + cooldown_meme
+
+	//GRAB INTENT
+	else if(findtext(command, "grab"))
+		for(var/mob/living/L in listeners)
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				H.a_intent_change(INTENT_GRAB)
+		next_command = world.time + cooldown_meme
+
+	//HARM INTENT
+	else if(findtext(command, "harm"))
+		for(var/mob/living/L in listeners)
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				H.a_intent_change(INTENT_HARM)
+		next_command = world.time + cooldown_meme
+
+	//THROW/CATCH
+	else if(findtext(command, "throw") || findtext(command, "catch"))
+		for(var/mob/living/L in listeners)
```

Comment:
Added a dead target check. If they're unconscious they can still listen, since there's a wake up and a heal command. Also i think that ordering a sleeping guy to sit down and him sleep-sitting on the chair is pretty funny.